lib/elements/clone.py

```diff
@@ -62,86 +53,106 @@ def clone(self):
            type='float')
     @cache
     def clone_fill_angle(self):
-        return self.get_float_param('angle') or None
+        return self.get_float_param('angle')
 
     @property
     @param('flip_angle',
            _('Flip angle'),
-           tooltip=_("Flip automatically calucalted angle if it appears to be wrong."),
+           tooltip=_(
+               "Flip automatically calculated angle if it appears to be wrong."),
            type='boolean')
     @cache
     def flip_angle(self):
-        return self.get_boolean_param('flip_angle')
+        return self.get_boolean_param('flip_angle', False)
 
     def get_cache_key_data(self, previous_stitch):
         source_node = get_clone_source(self.node)
-        source_elements = self.clone_to_element(source_node)
+        source_elements = self.clone_to_elements(source_node)
         return [element.get_cache_key(previous_stitch) for element in source_elements]
 
-    def clone_to_element(self, node):
+    def clone_to_elements(self, node):
         from .utils import node_to_elements
-        return node_to_elements(node, True)
-
-    def to_stitch_groups(self, last_patch=None):
-        patches = []
-
-        source_node = get_clone_source(self.node)
-        if source_node.tag not in EMBROIDERABLE_TAGS:
-            return []
-
-        old_transform = source_node.get('transform', '')
-        source_transform = source_node.composed_transform()
-        source_path = Path(source_node.get_path()).transform(source_transform)
-        transform = Transform(source_node.get('transform', '')) @ -source_transform
-        transform @= self.node.composed_transform() @ Transform(source_node.get('transform', ''))
-        source_node.set('transform', transform)
-
-        old_angle = float(source_node.get(INKSTITCH_ATTRIBS['angle'], 0))
-        if self.clone_fill_angle is None:
-            rot = transform.add_rotate(-old_angle)
-            angle = self._get_rotation(rot, source_node, source_path)
-            if angle is not None:
-                source_node.set(INKSTITCH_ATTRIBS['angle'], angle)
-        else:
-            source_node.set(INKSTITCH_ATTRIBS['angle'], self.clone_fill_angle)
-
-        elements = self.clone_to_element(source_node)
-        for element in elements:
-            stitch_groups = element.to_stitch_groups(last_patch)
-            patches.extend(stitch_groups)
-
-        source_node.set('transform', old_transform)
-        source_node.set(INKSTITCH_ATTRIBS['angle'], old_angle)
-        return patches
-
-    def _get_rotation(self, transform, source_node, source_path):
+        elements = []
+        if node.tag in EMBROIDERABLE_TAGS:
+            elements = node_to_elements(node, True)
+        elif node.tag == SVG_GROUP_TAG:
+            for child in node.iterdescendants():
+                elements.extend(node_to_elements(child, True))
+        return elements
+
+    def to_stitch_groups(self, last_patch=None) -> list[StitchGroup]:
```

Comment:
Same here for python 3.8 support I'm afraid we have to use List from typing (as well as further down in the file)

Reply:
Done.